cms/static/js/features_jsx/studio/CourseOrLibraryListing.jsx

```diff
@@ -3,86 +3,178 @@
-        props.items.map((item, i) =>
-          (
-            <li key={i} className="course-item" data-course-key={item.course_key}>
-              <a className={linkClass} href={item.url}>
-                <h3 className="course-title" id={`title-${idBase}-${i}`}>{item.display_name}</h3>
-                <div className="course-metadata">
-                  <span className="course-org metadata-item">
-                    <span className="label">{gettext('Organization:')}</span>
-                    <span className="value">{item.org}</span>
-                  </span>
-                  <span className="course-num metadata-item">
-                    <span className="label">{gettext('Course Number:')}</span>
-                    <span className="value">{item.number}</span>
-                  </span>
-                  { item.run &&
-                  <span className="course-run metadata-item">
-                    <span className="label">{gettext('Course Run:')}</span>
-                    <span className="value">{item.run}</span>
-                  </span>
-                  }
-                  { item.can_edit === false &&
-                  <span className="extra-metadata">{gettext('(Read-only)')}</span>
-                  }
-                </div>
-              </a>
-              { item.lms_link && item.rerun_link &&
-              <ul className="item-actions course-actions">
-                { allowReruns &&
-                <li className="action action-rerun">
-                  <a
-                    href={item.rerun_link}
-                    className="button rerun-button"
-                    aria-labelledby={`re-run-${idBase}-${i} title-${idBase}-${i}`}
-                    id={`re-run-${idBase}-${i}`}
-                  >{gettext('Re-run Course')}</a>
-                </li>
-                }
-                <li className="action action-view">
-                  <a
-                    href={item.lms_link}
-                    rel="external"
-                    className="button view-button"
-                    aria-labelledby={`view-live-${idBase}-${i} title-${idBase}-${i}`}
-                    id={`view-live-${idBase}-${i}`}
-                  >{gettext('View Live')}</a>
-                </li>
-                { can_archive &&
-                <li className="action action-view">
-                  <a
-                    href={item.archive_link}
-                    className="button view-button"
-                    aria-labelledby={`view-archive-${idBase}-${i} title-${idBase}-${i}`}
-                    id={`view-archive-${idBase}-${i}`}
-                  >{gettext('Archive')}</a>
-                </li>
-                }
-              </ul>
-              }
-            </li>
-          ),
-        )
-      }
-    </ul>
-  );
+        this.state = {
+            isOpen: false,
+            url: "",
+            text: ""
+        };
+
+        this.renderConfirmationDialog = this.renderConfirmationDialog.bind(this);
```

Comment:
Use Arrow Functions to avoid binding `this`.

Reply:
React in edx-platform is older version and doesn't support ES6. I tried using arrow functions first but `paver update_assets` fails to compile this `jsx` file. Also, all the other `jsx` files contain the same format.

Reply:
a plugin in .babelrc was missing. I have added it and changed these functions to arrow ones